sdk-java/src/main/java/io/littlehorse/sdk/worker/internal/RebalanceThread.java

```diff
@@ -104,6 +104,23 @@ public void onNext(RegisterTaskWorkerResponse response) {
                         removed.close();
                     }
                 }
+                LittleHorseGrpc.LittleHorseStub stub =
+                        config.getAsyncStub(runningConnection.getHost(), runningConnection.getPort());
+                // This loop replaces each PollThread that stops running with a fresh PollThread
+                for (int i = 0; i < runningConnections.get(runningConnection).size(); i++) {
+                    PollThread existingPollThread =
+                            runningConnections.get(runningConnection).get(i);
```

Comment:
[Suggestion]
Keep the number of gets as low as possible because this is a `ConcurrentHashMap` and might block other threads.